### ISSUE TITLE
[Bug] switch transformers 结构的 MixLoRA 中 router loss 无效

### ISSUE THREAD

**Issue by loadingyy** [mikecovlee/mLoRA#54]
gate模块根本没生效，一直在训练LoRA，gate的输入是在checkpoint中生成的，requires_grad是false，我通过给gate_weight挂钩子发现，gate从未被反向传播。模型有效果是因为lora在适应gate的初始权重，使moe有一些效果，并且aux-loss看似在降低，实际上aux-loss一点效果没有

**Comment by loadingyy** [comment]
对不起，我搞错了
